### PR TITLE
Update return codes in metadata

### PR DESCRIPTION
**Pull request by brandonforster**
#284 #409 #413

### CODE REVIEW

File: internal/core/metadata/rest_addressable.go [hunk 1]
```diff
@@ -52,7 +52,7 @@ func restAddAddressable(w http.ResponseWriter, r *http.Request) {
 	err := json.NewDecoder(r.Body).Decode(&a)
 	if err != nil {
 		LoggingClient.Error(err.Error(), "")
-		http.Error(w, err.Error(), http.StatusServiceUnavailable)
+		http.Error(w, err.Error(), http.StatusInternalServerError)
```
Comment (feclare) [inline internal/core/metadata/rest_addressable.go]: Should be bad request

File: internal/core/metadata/rest_deviceservice.go [hunk 2]
```diff
@@ -667,7 +667,7 @@ func restUpdateServiceOpStateByName(w http.ResponseWriter, r *http.Request) {
 	if !f {
 		err = errors.New("Invalid State: " + os + " Must be 'ENABLED' or 'DISABLED'")
 		LoggingClient.Error(err.Error(), "")
-		http.Error(w, err.Error(), http.StatusServiceUnavailable)
+		http.Error(w, err.Error(), http.StatusInternalServerError)
```
Comment (feclare) [inline internal/core/metadata/rest_deviceservice.go]: Should be bad request

File: internal/core/metadata/rest_deviceservice.go [hunk 3]
```diff
@@ -828,14 +828,14 @@ func restUpdateServiceLastReportedByName(w http.ResponseWriter, r *http.Request)
...
-		http.Error(w, err.Error(), http.StatusServiceUnavailable)
+		http.Error(w, err.Error(), http.StatusBadRequest)
 		return
 	}
 	var vlr string = vars[LASTREPORTED]
 	lr, err := strconv.ParseInt(vlr, 10, 64)
 	if err != nil {
 		LoggingClient.Error(err.Error(), "")
-		http.Error(w, err.Error(), http.StatusServiceUnavailable)
+		http.Error(w, err.Error(), http.StatusInternalServerError)
```
Comment (feclare) [inline internal/core/metadata/rest_deviceservice.go]: Should be bad request

**Comment by tsconn23** [comment]
@brandonforster Here's the error from the docs-verify job. Looks like "description:" is duplicated.
```
+ raml2html core-metadata.raml -o _build/html/core-metadata.html
[Error: mapping values are not allowed here
  in "core-metadata.raml", line 1548, column 41:
    ...           description: description: for malformed or unparsable req ...
```

File: api/raml/core-metadata.raml [hunk 4]
```diff
@@ -1425,28 +1523,32 @@ schemas:
...
+        description: Return ScheduleEvents matching given name (schedule names should be unique). May be null if no schedule events matches on the name provided. Returns Internal Service Error (HTTP 500) for unknown or unanticipated issues. NotFoundException (HTTP 404) if no ScheduleEvent is found with the provided name.
         responses:
             "200":
                 description: schedule event matching on name
                 body:
                     application/json:
                         schema: scheduleevent
                         example: '{"id":"57bb5837555e5218873e5a23","created":1471895607378,"modified":1471895971882,"origin":1471806386919,"name":"scrub pushed records","addressable":{"name":"scrub pushed records"},"parameters":null,"schedule":"daily","service":"edgex-support-scheduler"}'
+            "400":
+                description: description: for malformed or unparsable requests
```
Comment (tsconn23) [inline api/raml/core-metadata.raml]: @brandonforster Yep, here it is

File: internal/core/metadata/rest_addressable.go (hunk 1 above)
Comment (tsconn23) [reply internal/core/metadata/rest_addressable.go]: @brandonforster @feclare agreed

**Comment by tsconn23** [comment]
@brandonforster Requested changes were made, so this is good to go. Just rebase the PR branch please and once the verify jobs complete, will merge. Thanks.

**Pull request merged**